src/RokuDeploy.ts

```diff
@@ -731,6 +716,67 @@ export class RokuDeploy {
         return this.doPostRequest(deleteOptions);
     }
 
+    /**
+     * Gets a screenshot from the device. A side-loaded channel must be running or an error will be thrown.
+     */
+    public async takeScreenshot(options?: RokuDeployOptions & { screenshotPath?: string }) {
+        options.screenshotPath = options.screenshotPath ?? this.screenshotDir;
+        options = this.getOptions(options);
+
+        let saveFilePath: string;
+        const fileExtension = path.extname(options.screenshotPath);
+        // is this assumed to be a dir?
+        if (fileExtension === '') {
+            saveFilePath = path.join(options.screenshotPath, `screenshot-${dayjs().format('YYYY-MM-DD-HH.mm.ss.SSS')}.jpg`);
+
+            // is this a file path with jpg extension
+        } else if (/\.jp[e]?g/i.test(fileExtension)) {
+            saveFilePath = options.screenshotPath;
+
+            // everything else
+        } else {
+            throw new Error(`Can not, will not, convert image to ${fileExtension}`);
+        }
+
+        // Ask for the device to make an image
+        let createScreenshotResult = await this.doPostRequest({
+            ...this.generateBaseRequestOptions('plugin_inspect', options),
+            formData: {
+                mysubmit: 'Screenshot',
+                archive: ''
+            }
+        });
+
+        // Pull the image url out of the response body
+        const imageUrlOnDevice = /["'](pkgs\/dev\.jpg\?.+?)['"]/gi.exec(createScreenshotResult.body)?.[1];
```

Comment:
you can change the devices output from `jpg` to `png` I need to add support for this here and in the check for what format you asked for.

Reply:
Thanks @triwav for pointing this out. 
You can do this with:
home 5x -> up -> right -> down -> left -> up
and using `Cycle screenshot output format`